test/parallel/test-http-mutable-headers.js

```diff
@@ -21,6 +21,13 @@ const cookies = [
 const s = http.createServer(common.mustCall((req, res) => {
   switch (test) {
     case 'headers':
+      // Check that header-related functions work before setting any headers
+      // eslint-disable-next-line no-restricted-properties
+      assert.deepEqual(res.getHeaders(), {});
+      assert.deepStrictEqual(res.getHeaderNames(), []);
+      assert.deepStrictEqual(res.hasHeader('Connection'), false);
```

Comment:
Perhaps not the `hasHeader()`, but I think it would be good to leave it for object comparisons (including arrays).